src/vm/vm_core.rs

```diff
@@ -320,23 +311,17 @@ impl VirtualMachine {
 
     fn deduce_memory_cell(
         &mut self,
-        address: &MaybeRelocatable,
+        address: &Relocatable,
     ) -> Result<Option<MaybeRelocatable>, VirtualMachineError> {
-        if let MaybeRelocatable::RelocatableValue(addr) = address {
-            for (_, builtin) in self.builtin_runners.iter_mut() {
-                if builtin.base().segment_index == addr.segment_index {
-                    match builtin.deduce_memory_cell(address, &self.memory) {
-                        Ok(maybe_reloc) => return Ok(maybe_reloc),
-                        Err(error) => return Err(VirtualMachineError::RunnerError(error)),
-                    };
-                }
+        for (_, builtin) in self.builtin_runners.iter_mut() {
+            if builtin.base().segment_index == address.segment_index {
+                match builtin.deduce_memory_cell(&MaybeRelocatable::from(address), &self.memory) {
```

Comment:
To avoid using this `builtin.deduce_memory_cell(&MaybeRelocatable::from(address)` i made another PR